### PR TITLE
Migrate to ExoPlayer 2.11.5. Fixes #359

### LINKED ISSUE #359: Update to latest version of ExoPlayer

### PR DESCRIPTION
**Pull request by dturner**
Updating UAMP from ExoPlayer 2.10.0 to 2.11.5

**What are the changes?**
Blog post: https://medium.com/google-exoplayer/exoplayer-2-11-whats-new-e0e0701e4b6c
Relnotes: https://github.com/google/ExoPlayer/blob/release-v2/RELEASENOTES.md

**Changes which affect UAMP**
`setHandleAudioBecomingNoisy(true)`
Possibly: https://github.com/google/ExoPlayer/issues/6093
Possibly: https://github.com/google/ExoPlayer/issues/5891

**Issues**
Problem: MusicService.kt: (325, 9): 'onNotificationPosted' overrides nothing
Cause: Method signature changed from 

```
onNotificationPosted(
            notificationId: Int,
            notification: Notification?,
            ongoing: Boolean
        )
```

to

```
onNotificationPosted(
            notificationId: Int,
            notification: Notification,
            ongoing: Boolean
        )
```
Solution: Change Notification? to Notification

Problem: UampNotificationManager.kt: (88, 19): Class 'DescriptionAdapter' is not abstract and does not implement abstract member public abstract fun createCurrentContentIntent(p0: Player): PendingIntent? defined in com.google.android.exoplayer2.ui.PlayerNotificationManager.MediaDescriptionAdapter
Cause: Method signatures have changed
Solution: Update method signatures. See commit

Problem: MusicService.java:18: error: cannot find symbol
    private com.example.android.uamp.media.BecomingNoisyReceiver becomingNoisyReceiver;
                                          ^
Cause: Unknown - didn't investigate further because ExoPlayer can now handle the "becoming noisy" event for us. 
Solution: Remove class and add setHandleAudioBecomingNoisy(true) when SimpleExoPlayer is instantiated

Problem: MusicService.java:20: error: cannot find symbol
    private com.example.android.uamp.media.PackageValidator packageValidator;
Cause: The ctor sig is class PackageValidator(context: Context, @XmlRes xmlResId: Int) and the @XmlRes causes the build to fail
Solution: ?? clean and rebuild :/ 

Problem: UampPlaybackPreparer.kt: (41, 1): Class 'UampPlaybackPreparer' is not abstract and does not implement abstract member public abstract fun onPrepare(p0: Boolean): Unit defined in com.google.android.exoplayer2.ext.mediasession.MediaSessionConnector.PlaybackPreparer
Cause: method signature changes. E.g. onPrepareFromMediaId now supplies playWhenReady as a parameter
Solution: add the playWhenReady parameter to method signatures and pass this to exoPlayer where appropriate. Update comments. 

Problem: ExoPlayerFactory.newSimpleInstance is deprecated
Solution: 

```
SimpleExoPlayer.Builder(this).build().apply {
            setAudioAttributes(uAmpAudioAttributes, true)
            setHandleAudioBecomingNoisy(true)
            addListener(playerListener)
        }
```

Problem: PlayerNotificationManager.createWithNotificationChannel deprecated
Solution: Use the version which includes the channel description

Problem: .getWindow(windowIndex, window, true) deprecated
Solution: Change to .getWindow(windowIndex, window)

### CODE REVIEW

File: common/src/main/java/com/example/android/uamp/media/MusicService.kt [hunk 1]
```diff
@@ -68,7 +63,7 @@ import kotlinx.coroutines.launch
  * visit [https://developer.android.com/guide/topics/media-apps/audio-app/building-a-mediabrowserservice.html](https://developer.android.com/guide/topics/media-apps/audio-app/building-a-mediabrowserservice.html).
  */
 open class MusicService : MediaBrowserServiceCompat() {
-    private lateinit var becomingNoisyReceiver: BecomingNoisyReceiver
```
Comment (AquilesCanta) [inline common/src/main/java/com/example/android/uamp/media/MusicService.kt:71]: I may be wrong, but I think removing the instance variable is ok, because it's not accessed. However, the instance itself is doing something which is pause the content when the headphones are removed. You are the owner so you have the last word, but being a seemingly functional change, perhaps it's preferable to leave for a separate PR.

Comment (dturner) [reply common/src/main/java/com/example/android/uamp/media/MusicService.kt:71]: I think this is OK because ExoPlayer handles "becoming noisy" the same way.

According to the [release blog](https://medium.com/google-exoplayer/exoplayer-2-11-whats-new-e0e0701e4b6c) : 

"Playback will be paused automatically (setting playWhenReady to false) whenever the intent is received."

Comment (AquilesCanta) [reply common/src/main/java/com/example/android/uamp/media/MusicService.kt:71]: Ah! Amazing! Sorry for the spam.

**Comment by nic0lette** [comment]
Thank you!
